s3parq/fetch_parq.py

```diff
@@ -71,6 +77,35 @@
     Concat dataframes and return
 '''
 
+def fetch_max_partition_value(bucket: str, key: str, partition: str) -> any:
+    ''' Returns the max value of the specified partition
+    in the data type from the metadata.
+    '''
+    S3NamingHelper().validate_bucket_name(bucket_name=bucket)
+
+    all_files = _get_all_files_list(bucket=bucket, key=key)
+
+    partition_metadata = _get_partitions_and_types(
```

Comment:
what do you think of trimming it up a bit by grabbing only the datatype we care about and do the check right away? something like 
```
partition_datatype = _get_partitions_and_types(
        first_file_key=all_files[0], bucket=bucket)[partition]

if partition_datatype in NON_NUM_TYPES...
```
and then same for the values 
```
partition_values =     partition_values = _parse_partitions_and_values(
        file_paths=all_files, key=key)[partition]
```

Reply:
Due to the way the typing function is written, it takes dictionaries and maps the type conversions onto them. That function will need to be refactored in the future

Reply:
Addressed by #11 , resolving convo